sotodlib/site_pipeline/analyze_bright_ptsrc.py

```diff
@@ -290,214 +226,301 @@ def fit_params(
+    # peak amplitude plus fitted offsets
+    popt[1] = xi0 + popt_pointing[1] + popt[1]
+    popt[2] = eta0 + popt_pointing[2] + popt[2]
 
-    return all_params
+    return np.array([readout_id, *popt, snr])
 
 
-def make_maps(path, outdir, ofilename, sso_name, beamsize, n_modes):
-    ctx = core.Context(opj(path, "context", "context.yaml"))
-    obs_id = ctx.obsdb.get()[0]["obs_id"]
-    tod = ctx.get_obs(obs_id)
+def get_hw_positions(tele):
+    """ Get the hardware xi, eta, gamma positions and detector names
+    that belong to a specific tube."""
 
-    ofilename = opj(outdir, ofilename)
-    mask_dict = {"shape": "circle", "xyr": (0, 0, 5 * beamsize)}
+    from sotodlib.sim_hardware import sim_nominal, sim_detectors_toast
 
-    source_flags = planets.compute_source_flags(
-        tod=tod, mask=mask_dict, center_on=sso_name, res=0.02
-    )
+    freq_band = {
+        'SAT4': 'f030',
+        'SAT1': 'f090',
+        'SAT2': 'f090',
+        'SAT3': 'f230',
+        'LAT_o6': 'f030',
+        'LAT_i1': 'f090',
+        'LAT_i3': 'f090',
+        'LAT_i4': 'f090',
+        'LAT_i6': 'f090',
+        'LAT_i5': 'f230',
+        'LAT_c1': 'f230'}
+    hw = sim_nominal()
+    sim_detectors_toast(hw, tele)
 
-    ## Make the maps in degrees
-    ## mask ~ 5*beamsize
-    ## map size and mask radius could be user inputs
-    planets.make_map(
-        tod,
-        center_on=sso_name,
-        scan_coords=True,
-        n_modes=n_modes,
-        source_flags=source_flags,
-        size=10 * beamsize,
-        filename=ofilename,
-        comps="T",
-    )
+    qdr, det_names_hw = [], []
+    for names in hw.data['detectors'].keys():
+        if freq_band[tele] in names:
+            det_names_hw.append(names)
+            qdr.append([hw.data['detectors'][names]['quat'][3]] +
+                       list(hw.data['detectors'][names]['quat'][:3]))
+
+    quat_det = so3g.proj.quat.G3VectorQuat(np.array(qdr))
+    xi_hw, eta_hw, gamma_hw = so3g.proj.quat.decompose_xieta(quat_det)
+
+    return xi_hw, eta_hw, gamma_hw, np.array(det_names_hw)
+
+
+def plot_planet_footprints(tod, sso, tele, tube, **kwargs):
+    """ Plot the source's footprint on the focal plane """
+
+    fig = plt.figure(dpi=300)
+    ax = fig.add_subplot(111)
+
+    if 'xi_hw' in kwargs.keys() and 'eta_hw' in kwargs.keys():
+        xi_hw, eta_hw = kwargs['xi_hw'], kwargs['eta_hw']
+    else:
+        xi_hw, eta_hw, _, _ = get_hw_positions(tele)
+    ax.plot(np.degrees(xi_hw), np.degrees(eta_hw), '.')
+
+    csl = so3g.proj.CelestialSightLine.az_el(
+        tod.timestamps, tod.boresight.az, tod.boresight.el, weather="typical")
+    q_bore = csl.Q
+    planet = planets.SlowSource.for_named_source(sso, tod.timestamps[0] * 1.)
+    ra0, dec0 = planet.pos(tod.timestamps[0])
+    planet_q = so3g.proj.quat.rotation_lonlat(ra0, dec0)
+    q_total = ~q_bore * planet_q
+    xi_total, eta_total, _ = quat.decompose_xieta(q_total)
```

Comment:
it looks like nothing in this math accounts for boresight rotation of the telescope. (I'm noticing it while looking at these plots for a scan of Jupiter that has non-zero boresight).